Crash when compiling to run using gradlew.bat run

java.lang.ClassNotFoundException: net.demilich.metastone.game.spells.custom.MindGamesSpell09:28:27 ERROR - Trouble reading spell_mindgames.json

        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:264)
        at net.demilich.metastone.game.cards.desc.SpellDeserializer.deserialize(SpellDeserializer.java:28)
        at net.demilich.metastone.game.cards.desc.SpellDeserializer.deserialize(SpellDeserializer.java:16)
        at com.google.gson.TreeTypeAdapter.read(TreeTypeAdapter.java:58)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:103)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:196)
        at com.google.gson.Gson.fromJson(Gson.java:810)
        at com.google.gson.Gson.fromJson(Gson.java:875)
        at com.google.gson.Gson.fromJson(Gson.java:848)
        at net.demilich.metastone.game.cards.CardParser.parseCard(CardParser.java:74)
        at net.demilich.metastone.game.cards.CardCatalogue.loadCards(CardCatalogue.java:154)
        at net.demilich.metastone.game.cards.CardCatalogue.loadCards(CardCatalogue.java:145)
        at net.demilich.metastone.gui.cards.CardProxy.<init>(CardProxy.java:26)
        at net.demilich.metastone.ApplicationStartupCommand.execute(ApplicationStartupCommand.java:19)
        at net.demilich.nittygrittymvc.CommandObserverWrapper.handleNotification(CommandObserverWrapper.java:16)
        at net.demilich.nittygrittymvc.Facade.notifyObservers(Facade.java:79)
        at net.demilich.nittygrittymvc.Facade.sendNotification(Facade.java:177)
        at net.demilich.nittygrittymvc.Facade.sendNotification(Facade.java:166)
        at net.demilich.metastone.ApplicationFacade.startUp(ApplicationFacade.java:97)
        at net.demilich.metastone.MetaStone.start(MetaStone.java:45)
        at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$162(LauncherImpl.java:863)
        at com.sun.javafx.application.PlatformImpl.lambda$runAndWaitatformImpl.java:326)
        at com.sun.javafx.application.PlatformImpl.lambda$null$173(PlatformImpl.java:295)
        at java.security.AccessController.doPrivileged(Native Method)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$174(PlatformImpl.java:294)
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at com.sun.glass.ui.win.WinApplication.lambda$null$148(WinApplication.java:191)
        at java.lang.Thread.run(Thread.java:745)
Exception in Application start method
java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.sun.javafx.application.LauncherImpl.launchApplicationWithArgs(LauncherImpl.java:389)
        at com.sun.javafx.application.LauncherImpl.launchApplication(LauncherImpl.java:328)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sun.launcher.LauncherHelper$FXHelper.main(LauncherHelper.java:767)
Caused by: java.lang.RuntimeException: Exception in Application start method
        at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:917)
        at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$155(LauncherImpl.java:182)
        at java.lang.Thread.run(Thread.java:745)
Caused by: com.google.gson.JsonParseException: SpellDesc parser encountered an invalid spell class: net.demilich.metastone.game.spells.custom.MindGamesSpell
        at net.demilich.metastone.game.cards.desc.SpellDeserializer.deserialize(SpellDeserializer.java:31)
        at net.demilich.metaame.cards.desc.SpellDeserializer.deserialize(SpellDeserializer.java:16)
        at com.google.gson.TreeTypeAdapter.read(TreeTypeAdapter.java:58)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:103)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:196)
        at com.google.gson.Gson.fromJson(Gson.java:810)
        at com.google.gson.Gson.fromJson(Gson.java:875)
        at com.google.gson.Gson.fromJson(Gson.java:848)
        at net.demilich.metastone.game.cards.CardParser.parseCard(CardParser.java:74)
        at net.demilich.metastone.game.cards.CardCatalogue.loadCards(CardCatalogue.java:154)
        at net.demilich.metastone.game.cards.CardCatalogue.loadCards(CardCatalogue.java:145)
        at net.demilich.metastone.gui.cards.CardProxy.<init>(CardProxy.java:26)
        at net.demilich.metastone.ApplicationStartupCommand.execute(ApplicationStartupCommand.java:19)
        at net.demilich.nittygrittymvc.CommandObserverWrapper.handleNotification(CommandObserverWrapper.java:16)
        at net.demilich.nittygrittymvc.Facade.notifyObservers(Facade.java:79)
        at net.demilich.nittygrittymvc.Facade.sendNotification(Facade.java:177)
        at net.demilich.nittygrittymvc.Facade.sendNotification(Facade.java:166)
        at net.demilich.metastone.ApplicationFacade.startUp(ApplicationFacade.java:97)
        at net.demilich.metastone.MetaStone.start(MetaStone.java:45)
        at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$162(LauncherImpl.java:863)
        at com.sun.javafx.application.PlatformImpl.lambda$runAndWait$175(PlatformImpl.java:326)
        at com.sun.javafx.application.PlatformImpl.lambda$null$173(PlatformImpl.java:295)
        at java.security.AccessController.doPrivileged(Native Method)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$174(PlatformImpl.java:294)
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at com.sun.glass.ui.win.WinApplication.lambda$null$14plication.java:191)
        ... 1 more

That means your cards aren't updating. Delete your metastone folder or
modify the metastone properties files to reset your cards.

On Aug 11, 2016 8:35 AM, "swimmerwoad" <notifications@github.com> wrote:

> java.lang.ClassNotFoundException: net.demilich.metastone.game.
> spells.custom.MindGamesSpell09:28:27 ERROR - Trouble reading
> spell_mindgames.json
>
>     at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
>     at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
>     at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
>     at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
>     at java.lang.Class.forName0(Native Method)
>     at java.lang.Class.forName(Class.java:264)
>     at net.demilich.metastone.game.cards.desc.SpellDeserializer.deserialize(SpellDeserializer.java:28)
>     at net.demilich.metastone.game.cards.desc.SpellDeserializer.deserialize(SpellDeserializer.java:16)
>     at com.google.gson.TreeTypeAdapter.read(TreeTypeAdapter.java:58)
>     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:103)
>     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:196)
>     at com.google.gson.Gson.fromJson(Gson.java:810)
>     at com.google.gson.Gson.fromJson(Gson.java:875)
>     at com.google.gson.Gson.fromJson(Gson.java:848)
>     at net.demilich.metastone.game.cards.CardParser.parseCard(CardParser.java:74)
>     at net.demilich.metastone.game.cards.CardCatalogue.loadCards(CardCatalogue.java:154)
>     at net.demilich.metastone.game.cards.CardCatalogue.loadCards(CardCatalogue.java:145)
>     at net.demilich.metastone.gui.cards.CardProxy.<init>(CardProxy.java:26)
>     at net.demilich.metastone.ApplicationStartupCommand.execute(ApplicationStartupCommand.java:19)
>     at net.demilich.nittygrittymvc.CommandObserverWrapper.handleNotification(CommandObserverWrapper.java:16)
>     at net.demilich.nittygrittymvc.Facade.notifyObservers(Facade.java:79)
>     at net.demilich.nittygrittymvc.Facade.sendNotification(Facade.java:177)
>     at net.demilich.nittygrittymvc.Facade.sendNotification(Facade.java:166)
>     at net.demilich.metastone.ApplicationFacade.startUp(ApplicationFacade.java:97)
>     at net.demilich.metastone.MetaStone.start(MetaStone.java:45)
>     at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$162(LauncherImpl.java:863)
>     at com.sun.javafx.application.PlatformImpl.lambda$runAndWaitatformImpl.java:326)
>     at com.sun.javafx.application.PlatformImpl.lambda$null$173(PlatformImpl.java:295)
>     at java.security.AccessController.doPrivileged(Native Method)
>     at com.sun.javafx.application.PlatformImpl.lambda$runLater$174(PlatformImpl.java:294)
>     at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
>     at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
>     at com.sun.glass.ui.win.WinApplication.lambda$null$148(WinApplication.java:191)
>     at java.lang.Thread.run(Thread.java:745)
>
> Exception in Application start method
> java.lang.reflect.InvocationTargetException
> at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> at sun.reflect.NativeMethodAccessorImpl.invoke(
> NativeMethodAccessorImpl.java:62)
> at sun.reflect.DelegatingMethodAccessorImpl.invoke(
> DelegatingMethodAccessorImpl.java:43)
> at java.lang.reflect.Method.invoke(Method.java:498)
> at com.sun.javafx.application.LauncherImpl.launchApplicationWithArgs(
> LauncherImpl.java:389)
> at com.sun.javafx.application.LauncherImpl.launchApplication(
> LauncherImpl.java:328)
> at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> at sun.reflect.NativeMethodAccessorImpl.invoke(
> NativeMethodAccessorImpl.java:62)
> at sun.reflect.DelegatingMethodAccessorImpl.invoke(
> DelegatingMethodAccessorImpl.java:43)
> at java.lang.reflect.Method.invoke(Method.java:498)
> at sun.launcher.LauncherHelper$FXHelper.main(LauncherHelper.java:767)
> Caused by: java.lang.RuntimeException: Exception in Application start
> method
> at com.sun.javafx.application.LauncherImpl.launchApplication1(
> LauncherImpl.java:917)
> at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$155(
> LauncherImpl.java:182)
> at java.lang.Thread.run(Thread.java:745)
> Caused by: com.google.gson.JsonParseException: SpellDesc parser
> encountered an invalid spell class: net.demilich.metastone.game.
> spells.custom.MindGamesSpell
> at net.demilich.metastone.game.cards.desc.SpellDeserializer.
> deserialize(SpellDeserializer.java:31)
> at net.demilich.metaame.cards.desc.SpellDeserializer.
> deserialize(SpellDeserializer.java:16)
> at com.google.gson.TreeTypeAdapter.read(TreeTypeAdapter.java:58)
> at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(
> ReflectiveTypeAdapterFactory.java:103)
> at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$
> Adapter.read(ReflectiveTypeAdapterFactory.java:196)
> at com.google.gson.Gson.fromJson(Gson.java:810)
> at com.google.gson.Gson.fromJson(Gson.java:875)
> at com.google.gson.Gson.fromJson(Gson.java:848)
> at net.demilich.metastone.game.cards.CardParser.parseCard(
> CardParser.java:74)
> at net.demilich.metastone.game.cards.CardCatalogue.loadCards(
> CardCatalogue.java:154)
> at net.demilich.metastone.game.cards.CardCatalogue.loadCards(
> CardCatalogue.java:145)
> at net.demilich.metastone.gui.cards.CardProxy.(CardProxy.java:26)
> at net.demilich.metastone.ApplicationStartupCommand.execute(
> ApplicationStartupCommand.java:19)
> at net.demilich.nittygrittymvc.CommandObserverWrapper.handleNotification(
> CommandObserverWrapper.java:16)
> at net.demilich.nittygrittymvc.Facade.notifyObservers(Facade.java:79)
> at net.demilich.nittygrittymvc.Facade.sendNotification(Facade.java:177)
> at net.demilich.nittygrittymvc.Facade.sendNotification(Facade.java:166)
> at net.demilich.metastone.ApplicationFacade.startUp(
> ApplicationFacade.java:97)
> at net.demilich.metastone.MetaStone.start(MetaStone.java:45)
> at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$162(
> LauncherImpl.java:863)
> at com.sun.javafx.application.PlatformImpl.lambda$
> runAndWait$175(PlatformImpl.java:326)
> at com.sun.javafx.application.PlatformImpl.lambda$null$173(
> PlatformImpl.java:295)
> at java.security.AccessController.doPrivileged(Native Method)
> at com.sun.javafx.application.PlatformImpl.lambda$runLater$
> 174(PlatformImpl.java:294)
> at com.sun.glass.ui.InvokeLaterDispatcher$Future.
> run(InvokeLaterDispatcher.java:95)
> at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
> at com.sun.glass.ui.win.WinApplication.lambda$null$14plication.java:191)
> ... 1 more
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/demilich1/metastone/issues/216>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFsY_pmddzEoEK9dJiWp6LFVAb8Uxlcmks5qeyUagaJpZM4JiIDm>
> .
>

And by metastone folder, I mean the one in your Documents folder

On Aug 11, 2016 8:35 AM, "swimmerwoad" <notifications@github.com> wrote:

> java.lang.ClassNotFoundException: net.demilich.metastone.game.
> spells.custom.MindGamesSpell09:28:27 ERROR - Trouble reading
> spell_mindgames.json
>
>     at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
>     at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
>     at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
>     at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
>     at java.lang.Class.forName0(Native Method)
>     at java.lang.Class.forName(Class.java:264)
>     at net.demilich.metastone.game.cards.desc.SpellDeserializer.deserialize(SpellDeserializer.java:28)
>     at net.demilich.metastone.game.cards.desc.SpellDeserializer.deserialize(SpellDeserializer.java:16)
>     at com.google.gson.TreeTypeAdapter.read(TreeTypeAdapter.java:58)
>     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:103)
>     at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:196)
>     at com.google.gson.Gson.fromJson(Gson.java:810)
>     at com.google.gson.Gson.fromJson(Gson.java:875)
>     at com.google.gson.Gson.fromJson(Gson.java:848)
>     at net.demilich.metastone.game.cards.CardParser.parseCard(CardParser.java:74)
>     at net.demilich.metastone.game.cards.CardCatalogue.loadCards(CardCatalogue.java:154)
>     at net.demilich.metastone.game.cards.CardCatalogue.loadCards(CardCatalogue.java:145)
>     at net.demilich.metastone.gui.cards.CardProxy.<init>(CardProxy.java:26)
>     at net.demilich.metastone.ApplicationStartupCommand.execute(ApplicationStartupCommand.java:19)
>     at net.demilich.nittygrittymvc.CommandObserverWrapper.handleNotification(CommandObserverWrapper.java:16)
>     at net.demilich.nittygrittymvc.Facade.notifyObservers(Facade.java:79)
>     at net.demilich.nittygrittymvc.Facade.sendNotification(Facade.java:177)
>     at net.demilich.nittygrittymvc.Facade.sendNotification(Facade.java:166)
>     at net.demilich.metastone.ApplicationFacade.startUp(ApplicationFacade.java:97)
>     at net.demilich.metastone.MetaStone.start(MetaStone.java:45)
>     at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$162(LauncherImpl.java:863)
>     at com.sun.javafx.application.PlatformImpl.lambda$runAndWaitatformImpl.java:326)
>     at com.sun.javafx.application.PlatformImpl.lambda$null$173(PlatformImpl.java:295)
>     at java.security.AccessController.doPrivileged(Native Method)
>     at com.sun.javafx.application.PlatformImpl.lambda$runLater$174(PlatformImpl.java:294)
>     at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
>     at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
>     at com.sun.glass.ui.win.WinApplication.lambda$null$148(WinApplication.java:191)
>     at java.lang.Thread.run(Thread.java:745)
>
> Exception in Application start method
> java.lang.reflect.InvocationTargetException
> at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> at sun.reflect.NativeMethodAccessorImpl.invoke(
> NativeMethodAccessorImpl.java:62)
> at sun.reflect.DelegatingMethodAccessorImpl.invoke(
> DelegatingMethodAccessorImpl.java:43)
> at java.lang.reflect.Method.invoke(Method.java:498)
> at com.sun.javafx.application.LauncherImpl.launchApplicationWithArgs(
> LauncherImpl.java:389)
> at com.sun.javafx.application.LauncherImpl.launchApplication(
> LauncherImpl.java:328)
> at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> at sun.reflect.NativeMethodAccessorImpl.invoke(
> NativeMethodAccessorImpl.java:62)
> at sun.reflect.DelegatingMethodAccessorImpl.invoke(
> DelegatingMethodAccessorImpl.java:43)
> at java.lang.reflect.Method.invoke(Method.java:498)
> at sun.launcher.LauncherHelper$FXHelper.main(LauncherHelper.java:767)
> Caused by: java.lang.RuntimeException: Exception in Application start
> method
> at com.sun.javafx.application.LauncherImpl.launchApplication1(
> LauncherImpl.java:917)
> at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$155(
> LauncherImpl.java:182)
> at java.lang.Thread.run(Thread.java:745)
> Caused by: com.google.gson.JsonParseException: SpellDesc parser
> encountered an invalid spell class: net.demilich.metastone.game.
> spells.custom.MindGamesSpell
> at net.demilich.metastone.game.cards.desc.SpellDeserializer.
> deserialize(SpellDeserializer.java:31)
> at net.demilich.metaame.cards.desc.SpellDeserializer.
> deserialize(SpellDeserializer.java:16)
> at com.google.gson.TreeTypeAdapter.read(TreeTypeAdapter.java:58)
> at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(
> ReflectiveTypeAdapterFactory.java:103)
> at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$
> Adapter.read(ReflectiveTypeAdapterFactory.java:196)
> at com.google.gson.Gson.fromJson(Gson.java:810)
> at com.google.gson.Gson.fromJson(Gson.java:875)
> at com.google.gson.Gson.fromJson(Gson.java:848)
> at net.demilich.metastone.game.cards.CardParser.parseCard(
> CardParser.java:74)
> at net.demilich.metastone.game.cards.CardCatalogue.loadCards(
> CardCatalogue.java:154)
> at net.demilich.metastone.game.cards.CardCatalogue.loadCards(
> CardCatalogue.java:145)
> at net.demilich.metastone.gui.cards.CardProxy.(CardProxy.java:26)
> at net.demilich.metastone.ApplicationStartupCommand.execute(
> ApplicationStartupCommand.java:19)
> at net.demilich.nittygrittymvc.CommandObserverWrapper.handleNotification(
> CommandObserverWrapper.java:16)
> at net.demilich.nittygrittymvc.Facade.notifyObservers(Facade.java:79)
> at net.demilich.nittygrittymvc.Facade.sendNotification(Facade.java:177)
> at net.demilich.nittygrittymvc.Facade.sendNotification(Facade.java:166)
> at net.demilich.metastone.ApplicationFacade.startUp(
> ApplicationFacade.java:97)
> at net.demilich.metastone.MetaStone.start(MetaStone.java:45)
> at com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$162(
> LauncherImpl.java:863)
> at com.sun.javafx.application.PlatformImpl.lambda$
> runAndWait$175(PlatformImpl.java:326)
> at com.sun.javafx.application.PlatformImpl.lambda$null$173(
> PlatformImpl.java:295)
> at java.security.AccessController.doPrivileged(Native Method)
> at com.sun.javafx.application.PlatformImpl.lambda$runLater$
> 174(PlatformImpl.java:294)
> at com.sun.glass.ui.InvokeLaterDispatcher$Future.
> run(InvokeLaterDispatcher.java:95)
> at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
> at com.sun.glass.ui.win.WinApplication.lambda$null$14plication.java:191)
> ... 1 more
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/demilich1/metastone/issues/216>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFsY_pmddzEoEK9dJiWp6LFVAb8Uxlcmks5qeyUagaJpZM4JiIDm>
> .
>

--- a/game/src/main/java/net/demilich/metastone/game/cards/CardCatalogue.java
+++ b/game/src/main/java/net/demilich/metastone/game/cards/CardCatalogue.java
@@ -181,7 +181,7 @@ public class CardCatalogue {
 		String cardsCopiedWithVersion = MetastoneProperties.getProperty(CARDS_COPIED_PROPERTY);
 		boolean updateRequired;
 		try {
-			updateRequired = VersionInfo.updateRequired(BuildConfig.VERSION, cardsCopiedWithVersion);
+			updateRequired = VersionInfo.updateRequired(cardsCopiedWithVersion, BuildConfig.VERSION);
 		} catch(Exception e) {
 			// an exception will be thrown when stored 'cardsCopiedWithVersion' is null or invalid input,
 			// this is okay, just do an update in that case

